### PR TITLE
chore: usage of @commitlint/config-conventional

### PR DESCRIPTION
**Pull request by Divlo**
**What changes did you make? (Give an overview)**

- [x] Change `commitlint-config-awesome` to `@commitlint/config-conventional`
`commitlint-config-awesome` is less used/updated (not updated since Dec 16, 2019), break on `npm@7` because peerDependency don't match and we don't need it anymore because we don't use anymore `Greenkeeper`, also `@commitlint/config-conventional` is what it is used for `ts-standard`.

**Which issue (if any) does this pull request address?**

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: commitlint.config.js [hunk 1]
```diff
@@ -1 +0,0 @@
-module.exports = { extends: ['awesome'] }
```
Comment (mightyiam) [inline commitlint.config.js:1]: Keep the config in this file, unless there's a specific reason to change it.

File: package.json [hunk 2]
```diff
@@ -71,13 +71,12 @@
     "@typescript-eslint/eslint-plugin": ">=4.0.1"
   },
   "devDependencies": {
-    "@commitlint/cli": "11.0.0",
```
Comment (mightyiam) [inline package.json:74]: Don't update the version unless you have to, for the purpose of this PR (which is about replacing `commitlint-config-awesome` with `@commitlint/config-conventional`.

File: package.json [hunk 3]
```diff
@@ -71,13 +71,12 @@
...
-    "@commitlint/travis-cli": "11.0.0",
```
Comment (mightyiam) [inline package.json:75]: Don't remove this package, because it's still necessary. It's removal should be part of #530.

**Review by mightyiam — CHANGES_REQUESTED** [review]

File: commitlint.config.js (hunk 1 above)
Comment (Divlo) [reply commitlint.config.js:1]: I forgot to tell you why did I change this config file, I could tell you why in "Is there anything you'd like reviewers to focus on?" section of the PR.

Anyway here is the reason : I like consistency, and currently we have `.eslintrc.json`, so it feels better to do the same thing for this config file. If possible, we should try to be consistent in the codebase or for config files.

File: package.json (hunk 2 above)
Comment (Divlo) [reply package.json:74]: I don't think it is viable to open another PR just to update a version, it is still related to this PR even if it is indeed not necessary or for the purpose of this PR.

File: package.json (hunk 3 above)
Comment (Divlo) [reply package.json:75]: Sure, done! :smile:

File: commitlint.config.js (hunk 1 above)
Comment (mightyiam) [reply commitlint.config.js:1]: Perhaps. But this is not the PR for that change.

File: package.json (hunk 2 above)
Comment (mightyiam) [reply package.json:74]: I disagree. Related is not enough. PRs in this project are cheap. If you don't feel like making another PR—don't. But please don't include it with this one.

Comment (Divlo) [reply package.json:70]: Right, it's up to you! I will open a new PR if needed! :smile:

**Comment by Divlo** [comment]
Hey! :wave: @mightyiam 
Thanks again for taking your time to review my changes and excuse me for this long delay, I just updated to address your comments! Is there is anything that need to change before merging this PR ?

File: package.json [hunk 4]
```diff
@@ -59,21 +59,21 @@
     "eslint-config-standard": "^16.0.0"
   },
   "peerDependencies": {
+    "@typescript-eslint/eslint-plugin": ">=4.0.1",
```
Comment (mightyiam) [inline package.json:62]: Was this moved there automatically?

Comment (Divlo) [reply package.json:62]: Yes when I ran `npm install`.

Comment (mightyiam) [reply package.json:62]: That's an unrelated change, then. And it should be made prior to any other change made with npm. As a separate PR. No apologies from me on this. This is how I roll.

Comment (Divlo) [reply package.json]: Well, others people, when they'll do PRs will probably change that too I guess.
But fine like that, we'll see, I updated, so it's doesn't change these lines. :+1:

**Comment by mightyiam** [comment]
Thank you, @Divlo.

**Pull request merged**